### PR TITLE
fixed TextField of Map API isEmpty validation

### LINKED ISSUE #175: Bug: Textfield of 'Open Weather Map API' not checking empty string
### Description

In settings, Open Weather Map API TextField has no 'isEmpty' string-checking mechanism before identifying whether it's correct or wrong.

Desired output:
First It should be checked whether TextField is empty or not.

**Could you please assign this issue to me?**

### Screenshots

How to produce this issue:

https://github.com/CCExtractor/ultimate_alarm_clock/assets/69643676/50240708-3470-4338-9a7f-66efe2a70f2c

<br>
Device Name: Samsung S20 Ultra
<br>
Android Version: Android 13

### PR DESCRIPTION
**Pull request by prince02765**
### Description
I changed the TextField properties which can help to validate isEmpty before checking the correct API Key.

### Proposed Changes
- Validation is added
- Hint in textField added

## Screenshots

https://github.com/CCExtractor/ultimate_alarm_clock/assets/69643676/5d6f7506-6ea6-4e5c-be1a-6b794829f45d

## Checklist

- [x] Tests have been added or updated to cover the changes
- [x] Documentation has been updated to reflect the changes
- [x] Code follows the established coding style guidelines
- [x] All tests are passing

### CODE REVIEW

File: lib/app/modules/settings/views/weather_api.dart [hunk 1]
```diff
@@ -6,7 +6,7 @@ import 'package:ultimate_alarm_clock/app/utils/constants.dart';
 import 'package:ultimate_alarm_clock/app/utils/utils.dart';
 import 'package:url_launcher/url_launcher_string.dart';
 
-class WeatherApi extends StatelessWidget {
```
Comment (MarkisDev) [inline lib/app/modules/settings/views/weather_api.dart:9]: Why did you convert this to a stateful widget? We're using GetX for state management.

**Review by MarkisDev — CHANGES_REQUESTED** [review]
Please use GetX for state management, this issue should be relativelty easy to solve without having to convert it to a stateful class.

File: lib/app/modules/settings/views/weather_api.dart (hunk 1 above)
Comment (prince02765) [reply lib/app/modules/settings/views/weather_api.dart]: Actually, I'm newbie and don't know much about Flutter.

I apologize for that.

**Comment by prince02765** [comment]
@MarkisDev I converted to stateLessWidget on my second commit.

File: lib/app/modules/settings/views/weather_api.dart [hunk 2]
```diff
@@ -22,6 +22,7 @@ class WeatherApi extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    var _validate = false.obs;
```
Comment (MarkisDev) [inline lib/app/modules/settings/views/weather_api.dart:25]: Please move this to the controller and access it from there, we try to keep the UI and the business logic separate.

**Review by MarkisDev — CHANGES_REQUESTED** [review]

File: lib/app/modules/settings/views/weather_api.dart (hunk 2 above)
Comment (MarkisDev) [reply lib/app/modules/settings/views/weather_api.dart:25]: Also, great job! I assume you're new to GetX as well, so definitely well done!

Just remember to resolve merge conflicts as well!

**Comment by prince02765** [comment]
@MarkisDev I separated business logic and moved to the controller.

**Pull request merged**